students/liikt/main.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"bufio"
+	"encoding/csv"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"time"
+)
+
+func sanitize(s string) string {
+	return strings.ToLower(strings.Trim(s, "\n\r\t"))
+}
+
+func main() {
+	problemsFile := flag.String("path", "problems.csv", "This is the flag to the CSV containing the problems for the quiz")
+	timeout := flag.Int("timeout", 30, "The amount of time you have for a single question in seconds")
+	flag.Parse()
+
+	file, err := os.Open(*problemsFile)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer file.Close()
+
+	problems, err := csv.NewReader(file).ReadAll()
+	if err != nil {
+		log.Fatal(err)
+	}
+
```

Comment:
Your logic works perfect. It deals with the timeout while the program is still waiting for user input. However it's a bit complex. Here is a better way to orchestrate user input and timeouts:
```go

func main() {
	// until after you read the csv file
	correct := make(chan struct{})
	go query(problems, correct)
	answers := collect(correct)
	fmt.Printf("\nYou got %d out of %d answers.\n", answers, len(problems))
}

func query(problems [][]string, correct chan struct{}) {
	var inp string
	for c, q := range problems {
		fmt.Printf("Question %v: %v -> ", c, q[0])
		fmt.Scanln(&inp)
		if sanitize(inp) == sanitize(q[1]) {
			correct <- struct{}{}
		}
	}
}

func collect(correct chan struct{}) int {
	var counter int
	timer := time.NewTimer(time.Duration(timeout) * time.Second)
	for {
		select {
		case <-timer.C:
			return counter
		case <-correct:
			counter++
		}
	}
}
```
Notes:
- There is only one channel to deal with.
- The channel is only a signal and doesn't consume any space in memory.
- We are using a timer (NewTimer) instead of After. In this case both do fine, but a timer is more appropriate.
- Because the program ends here, we are fine. But in the code I wrote there is a goroutine leakage (calling query in a goroutine) and it doesn't know how to stop working. If your program runs longer than this, you have to deal with that too.